### PR TITLE
Add commands to verify built binary doesn't contian errors

### PR DESCRIPTION
**Pull request by derek-ho**
### Description

Related to #5952 - Dashboards verifies that it builds, but not that the built artifact can run. 

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: .github/workflows/build_and_test_workflow.yml [hunk 1]
```diff
@@ -422,6 +422,15 @@ jobs:
...
 
+      - name: Run `${{ matrix.name }}`
+        run: ./artifacts/target/${{ env.ARTIFACT_BUILD_NAME }}/bin/opensearch-dashboards &
+        shell: bash
+
+      - name: Health check 
+        run: |
+          timeout 300 bash -c 'while [[ "$(curl http://localhost:5601/api/status | jq -r '.status.overall.state')" != "green" ]]; do sleep 5; done'
+        shell: bash
+
```
Comment (derek-ho) [inline .github/workflows/build_and_test_workflow.yml:433]: Should catch issues like: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5952

**Comment by derek-ho** [comment]
@bandinib-amzn I think this PR should catch issues in the CI process in core

**Review by bandinib-amzn — APPROVED** [review]

File: .github/workflows/build_and_test_workflow.yml (hunk 1 above)
Comment (derek-ho) [reply .github/workflows/build_and_test_workflow.yml:433]: It seems like this workflow is not being run currently, so can't verify it works. @bandinib-amzn can you help me run it/can you enable it from your end just so we can verify?

**Review by manasvinibs — APPROVED** [review]

File: .github/workflows/build_and_test_workflow.yml (hunk 1 above)
Comment (AMoo-Miki) [reply .github/workflows/build_and_test_workflow.yml:433]: The linux artifact is used in BWC tests and I don't think OSD will come up "green" without an OS instance to connect to.

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]
The linux artifact is used in BWC tests and I don't think OSD will come up "green" without an OS instance to connect to.

**Comment by derek-ho** [comment]
> The linux artifact is used in BWC tests and I don't think OSD will come up "green" without an OS instance to connect to.

Ack. Would you accept a PR in which I spin up a simple opensearch instance and verify that the artifacts then respond to a simple health check?

**Comment by AMoo-Miki** [comment]
BWC already covers this. If a plugin needs to verify that their binaries are being build correctly, that test should be on that repo's workflows.

**Pull request closed without merging**